File: SexyBiscuit.Engine/Core/PlayMode.cs
namespace SexyBiscuit.Engine.Core;

/// <summary>
/// Whether gameplay is running. True in a standalone game; the editor turns it off outside
/// play mode, so actors that start in an edit-time scene (a loaded level, an undo, a hot reload)
/// do not run gameplay side effects such as a <see cref="Gameplay.GameMode"/> spawning players.
/// </summary>
/// <remarks>
/// Components still Awake and Start in edit mode so they can render and cache what they need;
/// only behaviour that changes the world belongs behind this flag. The editor sets it true just
/// before it loads the play scene and false again before it restores the edit scene.
/// </remarks>
public static class PlayMode
{
    public static bool IsActive { get; set; } = true;
}

File: html5/src/core/PlayMode.js
// -----------------------------------------------------------------------------
// PlayMode — whether the world is running or merely being edited.
// -----------------------------------------------------------------------------

/**
 * Components still receive `awake` and `start` in edit mode; only world-changing
 * behaviour — a GameMode spawning players, a spawner emitting actors — waits on
 * this flag, so opening a scene in the editor does not populate it.
 *
 * The standalone runtime leaves it true for the life of the process.
 */
export const PlayMode = {
    isActive: true,
};
